hil/client/base.py

```diff
@@ -60,9 +63,50 @@ def check_response(self, response):
             except ValueError:  # No JSON request body; typical
                                 # For methods PUT, POST, DELETE
                 return
-        else:
+        try:
             e = json.loads(response.content)
             raise FailedAPICallException(
                 error_type=e['type'],
                 message=e['msg'],
             )
+        # Catching 404's that do not return JSON
+        except ValueError:
+            return response.content
+
+
+def _find_reserved(string, slashes_ok=False):
+    """Returns a list of illegal characters in a string"""
+    if slashes_ok:
+        p = r"[^A-Za-z0-9 /$_.+!*'(),-]+"
+    else:
+        p = r"[^A-Za-z0-9 $_.+!*'(),-]+"
+    return list(x for l in re.findall(p, string) for x in l)
+
+
+def check_reserved(obj_type, obj_string, slashes_ok=False):
```

Comment:
I know this serves as a decorator and you have tested the functions with using this decorator. But do you guys think we need a direct unit test with this function? I don't have much experience. Any idea?

Reply:
My idea was to send some bad input and check to make sure it triggers the correct error as I did in `tests/unit/client.py`.  I'm not sure if it needs to be tested directly besides that; I'd like some input too.